### ISSUE TITLE
Block spammy users

### ISSUE THREAD

**Issue by chadyj** [status-im/status-react#6735]
[comment]: # (Please replace ... with your information. Remove < and >)

### User Story

As a user, I want to block users so that I can manage spammers.

### Description

Lately Status has seen a bunch of spam by users that post the same message in lots of channels such as:

> I've had a big failure in my investment.
> If you give me a bit of ETH I will surely succeed and pay you back twice.
> 0x3439c7d08463df4e2724ecdcf295a5a9f9fc54d1

or

> 0x62cede580c1e5900e4871c26ace86fb5ae15949c
> ETH please

#### Expected behavior
Can block users, and then not see their messages

### Solution
These type of messages can gunk up the status chat, so one way of dealing with this type of spam could be to block users and hide their messages.

This should work on both mobile and desktop.

Spam mitigation and prevention will have many approaches, but blocking users is just one small step.

**Comment by chadyj** [comment]
@yenda what did you have in mind for the UI?

@errorists has already done some work in WP1 that might be helpful. Maciej can you post it here? Either way you guys should sync :)

**Comment by yenda** [comment]
As discuss during our weekly chat meeting I want to open here the discussion on hard vs soft block. I think we can do the two separately if we have a clear semantic distinction.

# Hard block (aka permaban)

- no unblocking (at least for now)
- all messages from the user are deleted
- all future messages are ignored
- the quickest and easiest to implement: permabanned user are completely removed from contacts, stored aside as a list of blocked public-keys. Ignoring can be done on status-react side for now and later by status-go which could maintain the list of ignored public-keys in the future since there is already persistence for group-chats.

# Soft block (aka ignore)

- same feature familly as metadata I want to add to contacts, chats and messages
- when a user is marked as ignored, the metadata tag `ignored` is added to it's contact entry in the database. All his message are hidden in the UI, not counted in the unread counter, not shown in the last message of chats (these requirements are important, because right now it cannot be done in a straightforward way and requires lots of changes)
- a user can be `unignored` and all his messages are shown and counted again

**Comment by yenda** [comment]
Implemented and merged
